### PR TITLE
[INLONG-7867][Manager] Support data validation when importing Excel file

### LINKED ISSUE #7867: [Feature][Manager] Support data validation when importing Excel file
### Description

_No response_

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by featzhang**
- Fixes #7867 

### Validation result

image

```json
{
  "success": false,
  "errMsg": "Error in Row 3, Column 4:StreamField type must one of [date, string, double, float, int, long, timestamp];\nError in Row 5, Column 1:Value can not be empty!;",
  "data": null
}
```

### CODE REVIEW

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/util/Preconditions.java [hunk 1]
```diff
@@ -153,6 +155,17 @@ public static void expectNotBlank(String obj, ErrorCodeEnum errorCodeEnum, Strin
         }
     }
 
+    public static void expectBlank(String obj, ErrorCodeEnum errorCodeEnum, String errMsg) {
+        if (StringUtils.isNotBlank(obj)) {
+            throw new BusinessException(errorCodeEnum, errMsg);
+        }
+    }
+    public static void expectBlank(List<String> obj, ErrorCodeEnum errorCodeEnum, String errMsg) {
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/util/Preconditions.java:163]: expectEmpty.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java [hunk 2]
```diff
@@ -457,6 +477,27 @@ private static String parseCellValue(Cell cell) {
...
+     * Validate the cell value of a given field in the Excel sheet
+     *
+     * @param fieldMeta the meta information of the field to validate
+     * @param value     the value of the field to validate
+     */
+    private static Optional<String> validateCellValue(
+            FieldMeta fieldMeta,
+            Object value) {
+        ExcelCellValidator cellValidator = fieldMeta.getCellValidator();
+        if (cellValidator != null) {
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java:490]:
```
  if (cellValidator != null&&!cellValidator.validate(value)) {
       return Optional.of(cellValidator.getInvalidTip());
  } else {
        return Optional.empty();
  }
```

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java [hunk 3]
```diff
@@ -372,35 +373,54 @@ public static <E> List<E> read(InputStream is, Class<E> clazz)
...
+                                .append(colIndex + 1).append(":").append(info).append(";"));
                         fieldMeta.getField().setAccessible(true);
                         fieldMeta.getField().set(instance, value);
-
                     }
                     if (hasValueInRow) {
                         currentResult.add(instance);
                     }
+                    if (colValidateResult.length() > 0) {
+                        String lineValidateResult = "Error in Row " + (rowNum + 1) + ", " + colValidateResult;
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java:418]: Suggest use `String.format()`.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java [hunk 4]
```diff
@@ -553,17 +599,22 @@ public static <T> ClassMeta<T> of(Class<T> clazz)
...
                 if (excelField != null) {
+                    Class<? extends ExcelCellValidator> validatorClass = excelField.validator();
                     ExcelCellDataTransfer excelCellDataTransfer = excelField.x2oTransfer();
+                    ExcelCellValidator excelCellValidator = null;
+                    if (validatorClass != ExcelCellValidator.class) {
+                        excelCellValidator = validatorClass.newInstance();
+                    }
                     meta.addField(field.getName(), excelField.name(), field, field.getType(),
-                            excelCellDataTransfer);
+                            excelCellDataTransfer, excelCellValidator);
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java:609]: `excelCellValidator` maybe is null.

**Review by fuweng11 — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**